### PR TITLE
Add the vertex endpoint resource.

### PR DESCRIPTION
**Pull request by trodge**
Add the `google_vertex_ai_endpoint` resource.

part of https://github.com/hashicorp/terraform-provider-google/issues/9298

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_vertex_ai_endpoint
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Would you like any help with this change. Is it ready for review?

**Comment by trodge** [comment]
I would like some help cleaning up the service networking connections in the CI test project. The test is currently failing with:
```
Error: Error waiting for Create Service Networking Connection: Error code 9, message: You have reached the limit of number of tenant resources in the tenancy unit for service 'servicenetworking.googleapis.com' and consumer '1067888929963'.
```

**Comment by ScottSuarez** [comment]
I would take a look at the ci project and try and clear up some networking resources there.
https://pantheon.corp.google.com/welcome?project=ci-test-project-188019

**Comment by trodge** [comment]
> I would take a look at the ci project and try and clear up some networking resources there. https://pantheon.corp.google.com/welcome?project=ci-test-project-188019

I don't seem to have access.

**Comment by ScottSuarez** [comment]
Good to know ! Let me follow up with Riley ~ !

File: mmv1/products/vertexai/api.yaml [hunk 1]
```diff
@@ -96,13 +96,232 @@ objects:
...
+                        values:
+                          - :ACCELERATOR_TYPE_UNSPECIFIED
+                          - :NVIDIA_TESLA_K80
+                          - :NVIDIA_TESLA_P100
+                          - :NVIDIA_TESLA_V100
+                          - :NVIDIA_TESLA_P4
+                          - :NVIDIA_TESLA_T4
+                          - :NVIDIA_TESLA_A100
+                          - :TPU_V2
+                          - :TPU_V3
```
Comment (ScottSuarez) [inline mmv1/products/vertexai/api.yaml:192]: Lets change this to a string with a link to value types. 

Enum's are difficult to maintain

**Review by ScottSuarez — COMMENTED** [review]
nice ! looks good. Only a small documentation change

File: mmv1/products/vertexai/api.yaml (hunk 1 above)
Comment (trodge) [reply mmv1/products/vertexai/api.yaml:192]: What should the link to value types look like? Do you have an example?

Comment (ScottSuarez) [reply mmv1/products/vertexai/api.yaml]: sorry missed this comment ! I see you got it sorted on your most recent change. Resolving this.

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**